include/odp/api/spec/schedule_types.h

```diff
@@ -244,6 +244,235 @@ typedef struct odp_schedule_capability_t {
 
 } odp_schedule_capability_t;
 
+/**
+ * Region specific cache stashing configuration
+ *
+ * Region specific cache stashing configuration for different cache levels.
+ * Application can, for example, configure caching of certain portions of a
+ * region to L2 while configuring another portion to be cached to L3 or
+ * alternatively caching to both levels by configuring overlapping offsets and
+ * byte counts.
+ */
+typedef struct odp_cache_stash_region_t {
+	/** L2 cache stashing */
+	struct {
+		/** Byte offset into a region to start caching from
+		 *
+		 *  Depending on the implementation, this might be rounded down
+		 *  to a more suitable boundary.
+		 */
+		uint32_t offset;
+
+		/** Length in bytes to cache
+		 *
+		 *  Depending on the implementation, this might be rounded up
+		 *  to a more suitable boundary.
+		 */
+		uint32_t len;
+
+	} l2;
+
+	/** L3 cache stashing */
+	struct {
+		/** Byte offset into a region to start caching from
+		 *
+		 *  Depending on the implementation, this might be rounded down
+		 *  to a more suitable boundary.
+		 */
+		uint32_t offset;
+
+		/** Length in bytes to cache
+		 *
+		 *  Depending on the implementation, this might be rounded up
+		 *  to a more suitable boundary.
+		 */
+		uint32_t len;
+
+	} l3;
+
+} odp_cache_stash_region_t;
+
+/**
+ * Cache stashing configuration
+ *
+ * Cache stashing configuration for different data regions.
+ */
+typedef struct odp_cache_stash_config_t {
+	/** Region specific configuration toggle */
+	union {
+		/** Region bit fields */
+		struct {
+			/** Enable/disable event_metadata L2 cache stashing */
+			uint16_t event_metadata_l2  : 1;
+
+			/** Enable/disable event_metadata L3 cache stashing */
+			uint16_t event_metadata_l3  : 1;
+
+			/** Enable/disable event_data L2 cache stashing */
+			uint16_t event_data_l2      : 1;
+
+			/** Enable/disable event_data L3 cache stashing */
+			uint16_t event_data_l3      : 1;
+
+			/** Enable/disable event_user_area L2 cache stashing */
+			uint16_t event_user_area_l2 : 1;
+
+			/** Enable/disable event_user_area L3 cache stashing */
+			uint16_t event_user_area_l3 : 1;
+
+			/** Enable/disable queue_context L2 cache stashing */
+			uint16_t queue_context_l2   : 1;
+
+			/** Enable/disable queue_context L3 cache stashing */
+			uint16_t queue_context_l3   : 1;
+
+		};
+
+		/** All bits of the bit field structure
+		 *
+		 *  This field can be used to set/clear all bits, or bitwise
+		 *  operations over the entire structure.
+		 */
+		uint16_t all;
```

Comment:
To be more future proof, size of the field could be increased to 32 bits. 8/16 bits is already used and each new cache level (L1 or L4) would need 4 additional bits. There could be also other changes coming to flag usage (e.g. enable/disable/inherit).

Reply:
Bumped up to 32 bits in v5.